### PR TITLE
Adding sizeof definition for pyarrow tables and columns

### PR DESCRIPTION
**Pull request by rjzamora**
This PR adds (approximate) `sizeof` definitions for `pyarrow.lib.Table` and `pyarrow.lib.Column` objects. The new `sizeof_pyarrow_table` routine simply accounts for the size of the metadata, plus the cumulative size of all `pyarrow.Buffer` objects in the table. The new `sizeof_pyarrow_column` routine only accounts for the cumulative size of all `pyarrow.Buffer` objects in the column.

Note that these changes were motivated by discussion in [dask-cudf#44](https://github.com/rapidsai/dask-cuda/issues/44), and can be removed if/when [ARROW-6926](https://issues.apache.org/jira/browse/ARROW-6926) is directly addressed in pyarrow.

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/sizeof.py [hunk 1]
```diff
@@ -101,3 +101,27 @@ def sizeof_spmatrix_dok(s):
...
+
+    def _get_col_size(col):
+        p = 0
+        for chunk in col.data.iterchunks():
+            for buffer in chunk.buffers():
+                if buffer:
+                    p += buffer.size
+        return p
+
+    @sizeof.register(pa.lib.Table)
```
Comment (TomAugspurger) [inline dask/sizeof.py]: Is pyarrow.lib public? Or should this use `pa.Table`?

same question for `pa.lib.Column` below.

File: dask/sizeof.py [hunk 2]
```diff
@@ -101,3 +101,27 @@ def sizeof_spmatrix_dok(s):
     @sizeof.register(sparse.spmatrix)
     def sizeof_spmatrix(s):
         return sum(sizeof(v) for v in s.__dict__.values())
+
+
+@sizeof.register_lazy("pyarrow")
+def register_pyarrow():
+    import pyarrow as pa
+
+    def _get_col_size(col):
```
Comment (TomAugspurger) [inline dask/sizeof.py]: Question / style nitpick: can you put this implementation in the `sizeof_pyarrow_column` and just call `sizeof(col)` in the for loop for Table below?

```python
for col in table.itercolumns():
    p += sizeof(col)
```

I think that the registration happens first, so things should work out.

Comment (rjzamora) [reply dask/sizeof.py]: Right - I only avoided this, because sizeof_pyarrow_column is adding the 1000-byte serialization overhead for each column.

Comment (TomAugspurger) [reply dask/sizeof.py]: Ahh, that make sense.

**Comment by TomAugspurger** [comment]
Looks like CI is failing with a reference to `pa.lib.Column` not being found: https://travis-ci.org/dask/dask/jobs/601343047#L1057

```
>       @sizeof.register(pa.lib.Column)

        def sizeof_pyarrow_column(col):

E       AttributeError: module 'pyarrow.lib' has no attribute 'Column'
```

File: dask/sizeof.py (hunk 1 above)
Comment (rjzamora) [reply dask/sizeof.py]: Good call - my mistake

**Comment by TomAugspurger** [comment]
Is `pa.Column` gone in pyarrow 0.15?

```
In [1]: import pyarrow as pa

In [2]: pa.__version__
Out[2]: '0.15.0'

In [3]: pa.Column
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-16ff35b18c4d> in <module>
----> 1 pa.Column

AttributeError: module 'pyarrow' has no attribute 'Column'


```

That's causing a CI failure: https://travis-ci.org/dask/dask/jobs/601830315#L1057

**Comment by rjzamora** [comment]
> Is pa.Column gone in pyarrow 0.15?

Yeah, thanks for the note here - Looks like pyarrow 0.15 removed `Column` and directly replaced it with `ChunkedArray`

**Comment by mrocklin** [comment]
cc @ayushdg

**Comment by ayushdg** [comment]
Thanks @mrocklin for identifying the issue and @rjzamora for the fix!

**Comment by rjzamora** [comment]
@mrocklin - Do you feel that this `sizeof` implementation addresses our needs?

**Comment by TomAugspurger** [comment]
Skimming through https://github.com/rapidsai/dask-cuda/issues/44, this seems like it would solve things.

Only other comment is that in `distributed/protocol/arrow.py`, we handle RecordBatch. Should we add that to sizeof as well?

**Comment by mrocklin** [comment]
This seems good to me.  I'll merge tomorrow if there are no further comments (or anyone else please feel free to merge).

> Only other comment is that in distributed/protocol/arrow.py, we handle RecordBatch. Should we add that to sizeof as well?

Handling more things is good.  The current issue will be resolved with what's here.

**Comment by TomAugspurger** [comment]
Let's do this and followup with other arrow types as needed then. It may end up being unnecessary when they're done upstream.

**Comment by TomAugspurger** [comment]
Thanks @rjzamora!

**Pull request merged**